### DISCUSSION TITLE
make short url have whitespaces in it (example.com/my files.zip)

### DISCUSSION THREAD

**Discussion by ghost** [YOURLS/YOURLS#3694] [Q&A] [answered]
How would i make it so YOURLS can understand links like ``https://example.com/ my files.zip`` ? 

I know it'd be %20 but as far as i can see creating a short url link just deletes the space and it wont recognize it.

**Comment by dgw** [comment]
This is plugin territory and should be quite simple. A charset modification example for hyphens (`-`) ships as part of YOURLS: https://github.com/YOURLS/YOURLS/blob/master/user/plugins/hyphens-in-urls/plugin.php

[Forward slash](https://github.com/williambargent/YOURLS-Forward-Slash-In-Urls) and [full stop](https://github.com/SophiaAtkinson/YOURLS-full-stop-in-short-urls) have third party solutions listed in [the Awesome List](https://github.com/YOURLS/awesome) (though both are marked as archived, I've looked at the code and there's no apparent reason why they wouldn't still work).

**Reply by ghost** [reply]
I attempted what you said but no dice. 

1. plugin is enabled (bold signify this)
2. modified hyphen one to allow space, but I'm assuming I did something wrong
3. I attached code in case it'd help


```php                                
<?php
/**
 * Plugin Name: Allow Space in Short URLs
 * Plugin URI: http://yourls.org/
 * Description: Allow space in short URLs (like <tt>http://sho.rt/hello-world</tt>)
 * Version: 1.1
 * Author: Ozh
 * Author URI: http://ozh.org/
 */

// No direct call
if( !defined( 'YOURLS_ABSPATH' ) ) die();

// Add space to the allowed character set
yourls_add_filter( 'get_shorturl_charset', 'ozh_space_in_charset' );
// Unless we are crafting a random keyword
yourls_add_action('add_new_link_create_keyword', function() {yourls_remove_filter('get_shorturl_charset', 'ozh_space_in_charse>

function ozh_space_in_charset( $in ) {
        return $in.' ';
}
```

**Reply by ghost** [reply, marked as answer]
i just decided to go with another route. A dirty little edit from the plugin [yourls-case-insensitive](https://github.com/IMExperts/yourls-case-insensitive.git) i just made it strip %20 from the URL it receives  (``$keyword = str_replace('%20', '', $keyword)``)

**Reply by dgw** [reply]
I assume that the incorrect syntax (a long line ending in `>`) in your pasted code is due to copying it out of a terminal editor/pager. I've completed the cut-off line and tested it myself, finding that, indeed, ` ` is still removed from the custom keyword even when allowed by this very simple plugin, while a different special character like `*` or `&` works fine if added to the same `return` statement.

There seems to be some weirdness with how the admin panel (most likely its JavaScript) handles escaping certain characters. I've filed #3697 for something I found by accident along the way, and #3698 for what I think is directly affecting _this_ use case.

Meanwhile, your workaround to just ignore `%20` in the keyword during redirection is probably good enough for now. I'm glad you were able to find a way to get the functionality you wanted. 😸
